variables.tf

```diff
@@ -7,3 +7,125 @@ variable "resource_group_name" {
   type        = string
   description = "The name of the resource group in which to deploy the AVD-resources."
 }
+
+variable "data_location" {
+  type        = string
+  description = "The location to which metadata-resources will be deployed. This includes the host-pool, application group and workspace. Location of the virtual machines is defined by var.host_location. See [Data locations for Azure Virtual Desktop](https://docs.microsoft.com/en-us/azure/virtual-desktop/data-locations)."
+}
+
+variable "host_location" {
+  type        = string
+  description = "The location to which the hosts (VMs) will be deployed."
+}
+
+variable "vm_size" {
+  type        = string
+  description = "The size of the hosts. E.g. `Standard_D2s_v3`."
+}
+
+variable "subnet_id" {
+  type        = string
+  description = "The ID of the subnet to where the hosts will be deployed. Must be in the same region as var.host_location."
+}
+
+variable "host_source_image_reference" {
+  type = object({
+    publisher = string
+    offer     = string
+    sku       = string
+    version   = string
+  })
+  description = "The reference to the operating system that will be used in the hosts. You can find this with [az cli](https://docs.microsoft.com/en-us/azure/virtual-machines/linux/cli-ps-findimage) or [PowerShell](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/cli-ps-findimage). This should most likely be a variant of Windows 10/11 enterprise multi session-edition or Windows Server."
+
+  default = {
+    offer     = "windows-11"
+    publisher = "microsoftwindowsdesktop"
+    sku       = "win11-21h2-avd"
+    version   = "latest"
+  }
+}
+
+variable "local_admin_username" {
+  type        = string
+  description = "The username of the local admin account on the hosts."
+  default     = "azureuser"
+}
+
+variable "local_admin_password" {
+  type        = string
+  description = "The password of the local admin account on the hosts. Defaults to a randomly generated password. This will be saved in state."
```

Comment:
Mention that it defaults to randomly generated **unique** passwords for each host